### ISSUE TITLE
authorize.net to disable SSLv3 on Nov 4th

### ISSUE THREAD

**Issue by jasonfb** [activemerchant/active_merchant#1442]
We have been notified by Authorize.net that they will disable SSLv3 on November 4th. 

From their letter:
"This means that if your website or shopping cart solution uses SSLv3 to send transactions to Authorize.Net, you will no longer be able to process transactions." (full text here: https://gist.github.com/jasonfb/5eb3e724db448c83685e)
1. How can I make sure my active merchant Gem is using TSL? 
2. If it is using SSLv3, how can we switch it to using TSL? 

Although I have not dug into this deeply yet. I imagine this is a low-level configuration change. Any tips on places I should begin looking would be appreciated.

**Comment by joevandyk** [comment]
Authorize.net says they are disabling TLS 1.0, that only TLS >= 1.1 will be supported. How can we verify if we're using the right one?

**Comment by girasquid** [comment]
@joevandyk to test, run `openssl s_client -connect api2.authorize.net:443` and look at the value of `protocol` in the output. If you want to explicitly test TLS 1.1 or 1.2, you can add the `-tls1_1` or `-tls1_2` flags.

**Comment by brandoncc** [comment]
Thanks for that @girasquid. Do all versions of ActiveMerchant support TLS v1.2? I am working on an app which uses ActiveMerchant v1.4.1. Using your method above I connected successfully from my server using TLS v1.2, I just need to make sure that my version of ActiveMerchant will support it as well.

**Comment by girasquid** [comment]
@brandoncc ActiveMerchant is going to use your system SSL libraries - so if your server is good, you should be ok. Your ActiveMerchant version is **very old** though (about 8 years); I'd recommend updating. Authorize.net [is moving their URLs to Akamai](https://www.authorize.net/support/akamaifaqs/#why), and the newer versions will have the correct URL.

**Comment by brandoncc** [comment]
@girasquid I agree. Honestly I am a bit afraid of upgrading things because there is basically no test suite so I am afraid of breaking things. Most current gems don't support Ruby 1.8.7 which the app is running on. I don't see a list of requirements for ActiveMerchant, does it support Ruby 1.8.7 and Rails 2.3?

**Comment by girasquid** [comment]
Not anymore, sorry - Ruby 2.0+ and Rails 3.2+.

**Comment by brandoncc** [comment]
I figured that was the case. The app may get updated but that is out of scope at the moment. Thanks for your help :smile:  As for the Akamai urls, they will automatically forward the current urls as of June, so we are going to rely on that.

**Comment by baraka2000** [comment]
I found this old thread. I'm hoping someone can help me. I have an old Spree Commerce store that uses activemerchant 1.28.0, and rails 3.2.x and ruby 2.0 I think. Apparently Authorize.net is disabling support for tls 1.1, allowing only for 1.2. Anyone know if my version of activemerchant is using 1.2? I noticed @girasquid said it's a matter of the server running the app?? So I'm a little confused how to check? Any help here is appreciated.
